CDAP-14839 move table spi definitions to separate module

albertshau opened the pull request:
Moved StructuredTable and other relevant classes into a separate
cdap-storage-spi module with minimal dependencies.

albertshau committed:
CDAP-14839 move table spi definitions to separate module

Moved StructuredTable and other relevant classes into a separate
cdap-storage-spi module with minimal dependencies.

poornachandra commented:
Changes look good, can you also move the file StructuredTableTest to the new module?

albertshau committed:
CDAP-14839 move StructuredTableTest to the spi module

Move the abstract base class for StructuredTables into the spi
module and relevant pom changes to create a test-jar. Removed
usage of guava in the test.

albertshau commented:
https://builds.cask.co/browse/CDAP-DUT6774
